OpenMEEGConfigure.h.in

```diff
@@ -16,10 +16,10 @@ static const char version[] = "@OpenMEEG_VERSION@";
 
     #if _OPENMP>=201811 || __GNUC__>=9
         #define OPENMP_RANGEFOR
-        #define OPENMP_ITERATOR
+        // #define OPENMP_ITERATOR
```

Comment:
@papadop I needed to do this as otherwise I get:

```
[ 28%] Building CXX object OpenMEEG/CMakeFiles/OpenMEEG.dir/src/assembleHeadMat.cpp.o
In file included from /Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:48:
/Users/alex/work/src/openmeeg/OpenMEEG/include/operators.h:321:79: error: 'this' argument to member function 'end' has type 'const OpenMEEG::Triangles' (aka 'const vector<OpenMEEG::Triangle>'), but function is not marked const
                for (Triangles::const_iterator tit2=m2_triangles.begin();tit2<m2_triangles.end();++tit2) {
                                                                              ^~~~~~~~~~~~
/Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:125:17: note: in instantiation of function template specialization 'OpenMEEG::operatorS<OpenMEEG::SymMatrix>' requested here
                operatorS(mesh1,mesh2,symmatrix,orientation*inv_cond*K,gauss_order);
                ^
/Users/alex/miniconda3/bin/../include/c++/v1/vector:625:54: note: 'end' declared here
    _LIBCPP_INLINE_VISIBILITY iterator               end() _NOEXCEPT;
                                                     ^
In file included from /Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:48:
/Users/alex/work/src/openmeeg/OpenMEEG/include/operators.h:352:71: error: 'this' argument to member function 'end' has type 'const OpenMEEG::Triangles' (aka 'const vector<OpenMEEG::Triangle>'), but function is not marked const
        for (Triangles::const_iterator tit1=m1_triangles.begin();tit1<m1_triangles.end();++tit1) {
                                                                      ^~~~~~~~~~~~
/Users/alex/work/src/openmeeg/OpenMEEG/include/operators.h:376:13: note: in instantiation of function template specialization 'OpenMEEG::operatorD<OpenMEEG::SymMatrix>' requested here
            operatorD(m2, m1, mat, coeff, gauss_order);
            ^
/Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:134:17: note: in instantiation of function template specialization 'OpenMEEG::operatorD<OpenMEEG::SymMatrix>' requested here
                operatorD(mesh1,mesh2,symmatrix,Dcoeff,gauss_order,false);
                ^
/Users/alex/miniconda3/bin/../include/c++/v1/vector:625:54: note: 'end' declared here
    _LIBCPP_INLINE_VISIBILITY iterator               end() _NOEXCEPT;
                                                     ^
In file included from /Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:48:
/Users/alex/work/src/openmeeg/OpenMEEG/include/operators.h:265:83: error: 'this' argument to member function 'end' has type 'const OpenMEEG::Triangles' (aka 'const vector<OpenMEEG::Triangle>'), but function is not marked const
                    for (Triangles::const_iterator tit2=m2_triangles.begin();tit2<m2_triangles.end();++tit2) {
                                                                                  ^~~~~~~~~~~~
/Users/alex/work/src/openmeeg/OpenMEEG/src/assembleHeadMat.cpp:143:13: note: in instantiation of function template specialization 'OpenMEEG::operatorN<OpenMEEG::SymMatrix>' requested here
            operatorN(mesh1,mesh2,symmatrix,Ncoeff,gauss_order);
            ^
/Users/alex/miniconda3/bin/../include/c++/v1/vector:625:54: note: 'end' declared here
    _LIBCPP_INLINE_VISIBILITY iterator               end() _NOEXCEPT;
                                                     ^
3 errors generated.
make[2]: *** [OpenMEEG/CMakeFiles/OpenMEEG.dir/src/assembleHeadMat.cpp.o] Error 1
make[1]: *** [OpenMEEG/CMakeFiles/OpenMEEG.dir/all] Error 2
make: *** [all] Error 2
```
I dit not investigate further

Reply:
What is the compiler ?? gcc (that would seem strange), clang (that would seem strange too) or apple modified clang pretending to be gcc (which I think they do). Because _OPENMP>=201811 corresponds to OpenMP 5.0, which is supposed to support range for loops, so my guess is that this is not the part of the test which is satisfied.
https://www.openmp.org/wp-content/uploads/OpenMP-API-Specification-5.0.pdf

We need to find a way to select the right branch (the second one) of the preprocessor test.
Probably, it needs to refer to sthg like (__GNUC__>=9 && !__APPLE__) . This should work, but we would need to verify for future apple compilers.

Reply:
Another possibility would be to write cmake test to check whether range for/iterators loops are supported. This is probably a safer approach... and we already have all the infrastructure for that. It is just a matter of writting 2-3 tests for the different OpenMP variants.